File: backend/src/main/java/com/sunrisedental/service/AppointmentService.java
package com.sunrisedental.service;

import com.sunrisedental.dto.AppointmentRequest;
import com.sunrisedental.dto.AppointmentResponse;
import com.sunrisedental.dto.PagedResponse;

import java.time.LocalDate;
import java.time.LocalTime;

public interface AppointmentService {

    AppointmentResponse create(AppointmentRequest request);

    PagedResponse<AppointmentResponse> getAll(
            String search,
            LocalDate date,
            Long patientId,
            Long dentistId,
            Long treatmentId,
            String status,
            int page,
            int size,
            String sortBy,
            String sortDir);

    AppointmentResponse getById(Long id);

    AppointmentResponse update(Long id, AppointmentRequest request);

    AppointmentResponse cancel(Long id);

    boolean checkDentistAvailability(Long dentistId, LocalDate date, LocalTime time, Integer duration, Long excludeId);

    AppointmentSummary getSummary();

    record AppointmentSummary(
            long totalScheduled,
            long totalConfirmed,
            long totalPending,
            long totalCancelled
    ) {}
}

File: frontend/src/services/appointmentService.js
import api from './api';

/**
 * Service to handle appointment APIs.
 */
const appointmentService = {
  
  /**
   * Get all appointments (paginated and filtered).
   */
  getAll({ search = '', date = '', patientId = '', dentistId = '', treatmentId = '', status = '', page = 0, size = 10, sortBy = 'appointmentDate', sortDir = 'desc' } = {}) {
    const params = { search, status, page, size, sortBy, sortDir };
    if (dentistId) params.dentistId = dentistId;
    if (treatmentId) params.treatmentId = treatmentId;
    if (date) params.date = date;
    if (patientId) params.patientId = patientId;
    
    return api.get('/appointments', { params });
  },

  /**
   * Get appointment by ID.
   */
  getById(id) {
    return api.get(`/appointments/${id}`);
  },

  /**
   * Create a new appointment.
   */
  create(data) {
    return api.post('/appointments', data);
  },

  /**
   * Update an existing appointment.
   */
  update(id, data) {
    return api.put(`/appointments/${id}`, data);
  },

  /**
   * Cancel an appointment.
   */
  cancel(id) {
    return api.put(`/appointments/${id}/cancel`);
  },

  /**
   * Check if dentist is available for a time slot.
   */
  checkAvailability(dentistId, date, time, duration, excludeId = null) {
    const params = { dentistId, date, time, duration };
    if (excludeId) params.excludeId = excludeId;
    return api.get('/appointments/dentist-availability', { params });
  },

  /**
   * Get appointment statistics summary.
   */
  getSummary() {
    return api.get('/appointments/summary');
  }
};

export default appointmentService;
